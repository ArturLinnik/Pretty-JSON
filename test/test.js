// Require Node.js Dependencies
const { spawn } = require("child_process");
const { join } = require("path");

// Require Third-party Dependencies
const is = require("@slimio/is");

// Require Internal Dependencies
const prettyJSON = require("../");

async function executeNodeScript(path) {
    const { stdout } = spawn(process.argv[0], [path]);
    let str = "";
    for await (const buf of stdout) {
        str += buf;
    }

    str | 0;
    return str;
}

test("prettyJSON must be a function", () => {
    expect(is.func(prettyJSON)).toStrictEqual(true);
});

test("prettyJSON of non-object must return undefined", () => {
    expect(prettyJSON("hello")).toStrictEqual(undefined);
});

test("prettyJSON stdout must be the same", async() => {
    const stdout = await executeNodeScript(join(__dirname, "fixtures", "01.js"));
    expect(stdout).toMatchSnapshot();
});
