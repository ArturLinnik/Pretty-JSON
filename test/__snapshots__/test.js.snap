// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettyJSON stdout must be the same 1`] = `
"[1m[37m foo:   [39m[22m[32m'bar'[39m
[1m[37m hello: [39m[22m[32m'world'[39m
[1m[37m arr:   [39m[22m[90m[[39m
    [36m1[39m[90m, [39m[36m2[39m[90m, [39m[36m3[39m
 [90m][39m
"
`;
